File: stringReverseLeft.cpp
#include <iostream>
#include <string>

using namespace std;

string reverseLeftKcharToTheEnd(string s, int k) {
	string res = "";
	for (int i = k; i < s.length() + k; i++) {
		res.push_back(s[i % s.length()]);
	}
	return res;
}

int main() {
	string input;
	int k;
	cin >> input;
	cin >> k;
	cout << reverseLeftKcharToTheEnd(input, k) << endl; 
	return 0;
}

File: stringReverseLeft.py
def reverseLeftKcharToTheEnd(s, k):
	res = ""
	for i in range(k, k+len(s)):
		res += s[i % len(s)]
	return res

x = input()
y = input()

print(reverseLeftKcharToTheEnd(str(x),int(y)))
